### PR TITLE
[TF][GPU]: Enable bitwise shift test in TF.

### PR DESCRIPTION
**Pull request by pkowalc1**
### Details:
 - Enabled TF tests for Bitwise shift ops on GPU

This PR depends on #26251, which adds support for int16 and uint16 data types in elementwise shift op in gpu plugin. 
Also depends on #26383, which fixes a bug related to i16 support for elementwise ops in GPU.

### CODE REVIEW

**Review by rkazants — APPROVED** [review]

**Review by mitruska — APPROVED** [review]
The list of tested input types has been extended to:
[np.int8, np.int16, np.int32, np.int64, np.uint8, np.uint16, np.uint32, np.uint64])

**Review by rkazants — CHANGES_REQUESTED** [review]
Please fix

```
RuntimeError: Exception from src/inference/src/cpp/core.cpp:104:
Exception from src/inference/src/dev/plugin.cpp:53:
Check 'false' failed at src/plugins/intel_gpu/src/plugin/program_builder.cpp:185:
[GPU] ProgramBuilder build failed!
Program build failed(0_part_0):
4:1949:2: error: invalid operands to binary expression ('float' and 'float')
 DO_ELTWISE;
 ^~~~~~~~~~
4:1780:2: note: expanded from macro 'DO_ELTWISE'
....
----------------------------- Captured stdout call -----------------------------
OpenVINO version: 2024.4.0-16450-894fd44f4df-refs/pull/26194/head
Creating OV Core Engine...
Reading network files
Loading network
```

File: tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py [hunk 1]
```diff
@@ -61,15 +61,13 @@ def create_bitwise_net(self, x_shape, y_shape, is_y_const, input_type, op_type):
     @pytest.mark.parametrize('x_shape', [[4], [3, 4], [1, 2, 3, 4]])
     @pytest.mark.parametrize('y_shape', [[1], [4], [2, 3, 4]])
     @pytest.mark.parametrize('is_y_const', [True, False])
-    @pytest.mark.parametrize('input_type', [np.int8, np.int16, np.int32, np.int64,
-                                            np.uint8, np.uint16, np.uint32, np.uint64])
+    @pytest.mark.parametrize('input_type', [np.int8, np.int32, np.int64,
+                                            np.uint8, np.uint32])
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:65]: please revert this change. We should cover all possible type.

**Review by rkazants — CHANGES_REQUESTED** [review]

File: tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py (hunk 1 above)
Comment (pkowalc1) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:65]: The GPU plugin does not have uint64 data type implemented. Currently bitwise shift in GPU plugin does not support also in16 and uint16, but I am adding it.

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:65]: Please skip these cases properly and indicate a ticket number for adding these types for CPU:
```
if ie_device == 'GPU' and input_type in [...]:
    pytest.skip('...')
```

Comment (pkowalc1) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py]: Fixed.

File: tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py [hunk 2]
```diff
@@ -68,8 +68,8 @@ def create_bitwise_net(self, x_shape, y_shape, is_y_const, input_type, op_type):
     @pytest.mark.nightly
     def test_bitwise(self, x_shape, y_shape, is_y_const, input_type, op_type, ie_device, precision, ir_version,
                      temp_dir, use_legacy_frontend):
-        if ie_device == 'GPU':
-            pytest.skip("149424: Bitwise ops are not supported on GPU")
+        if ie_device == 'GPU' and input_type in [np.uint64]:
+            pytest.skip("149424: uint64 type is not supported on GPU")
```
[the remark this answers is not in the document]
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:72]: @pkowalc1, please create separate ticket for uint64 support on GPU and list this ticket number here

[the remark this answers is not in the document]
Comment (pkowalc1) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:72]: @rkazants if you think there is a real need for uint64 support on GPU, please create a ticket and motivate it properly(e.g. maybe there is a business need). Currently, uint64 on GPU will be converted to int32 (info from V. Paramuzow), so there is a support but it it will not work properly in this case.

[the remark this answers is not in the document]
Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:72]: The motivation is to be fully aligned with framework cases and capabilities. Our intention is to support not only known models but all other models about which we don't know.
Please create ticket and list here if you are responsible for resolving this task and if you want to change TF FE test sources. Otherwise, I can switch on these tests on GPU in a separate PR and create ticket on my own.

[the remark this answers is not in the document]
Comment (pkowalc1) [reply tests/layer_tests/tensorflow_tests/test_tf_BitwiseShift.py:72]: @rkazants This PR is ready to be merged - all types except u64(as discussed above) should be working fine. However, if you want to create your own PR - no problem, please just close this one.

**Review by rkazants — APPROVED** [review]
